cluster/cluster.yaml

```diff
@@ -1473,6 +1474,16 @@ Resources:
                 - ''
                 - - 'arn:aws:iam::{{.Cluster.InfrastructureAccount | getAWSAccountID}}:role/'
                   - !Ref MasterIAMRole
+          {{ else }}
+          - Action:
+              - 'sts:AssumeRoleWithWebIdentity'
+            Effect: Allow
+            Principal:
+              Federated: "arn:aws:iam::{{ .Cluster.InfrastructureAccount | getAWSAccountID }}:oidc-provider/{{.Cluster.LocalID}}.{{.Values.hosted_zone}}"
+            Condition:
+              StringLike:
+                "{{ .Cluster.LocalID }}.{{ .Values.hosted_zone }}:sub": "system:serviceaccount:visibility:logging-agent"
+          {{ end }}
```

Comment:
There's a race condition here because the role is updated before the logging-agent Deployment is updated to use the OIDC approach. All logging-agent Pods are rolled and during the update. Old Pods are still using kube2iam's approach but this role is already updated and the permissions are dropped such that kube2iam's approach doesn't work anymore. So it might be safer to just leave both `assumeRole` permissions in there during the migration phase.

Reply:
thanks, updated.